source/hash_table.c

```diff
@@ -496,6 +496,38 @@ int aws_hash_table_create(
     return AWS_OP_SUCCESS;
 }
 
+AWS_COMMON_API
+int aws_hash_table_put(struct aws_hash_table *map, const void *key, void *value, int *was_created) {
+    struct hash_table_state *state = map->p_impl;
+    struct aws_hash_element *p_elem;
+    int was_created_fallback;
+
+    if (!was_created) {
+        was_created = &was_created_fallback;
+    }
+
+    /* We might end up resizing the table, so stash the destroy fns now */
```

Comment:
I don't understand why these functions needed to be stashed here.

Reply:
I guess I can move the state initialization down below the create call.